File: py-rx-boto3/update.py
class Update(object):
    def __init__(self):
        self.update_count = 0

    def execute(self, db, table, stats):
        self.update_count += 1

        db.update_item(
            TableName=table,
            Key={
                'date': stats.date,
                'domain': stats.domain
            },
            UpdateExpression="ADD msg_count :c, msg_size :ms, archived_size :as",
            ExpressionAttributeValues={
                ':c': stats.count,
                ':ms': stats.size,
                ':as':  stats.archived_size
            },
            ReturnValues='NONE')

        return stats.ids

File: js-rx-promise/lib/update.js
var Promise = require('promise');

module.exports = function() {
  module.exports.prototype.updateCount = 0;

  module.exports.prototype.exec = function(db, table, stats) {
    var self = this;
    return new Promise(function(resolve, reject) {
      self.updateCount++;
      db.updateItem(makeParams(table, stats), function(err, data) {
        if (err) {
          console.warn(err, err.stack);
          reject(err);
        } else {
          resolve(stats.ids);
        }
      });
    });
  };
};

function makeParams(table, stats) {
  return {
    Key: {
      date: {
        'S': stats.date
      },
      domain: {
        'S': stats.domain
      }
    },
    TableName: table,
    UpdateExpression: 'ADD msg_count :c, msg_size :ms, archived_size :as',
    ExpressionAttributeValues: {
      ':c': {
        'N': stats.count.toString()
      },
      ':ms': {
        'N': stats.size.toString()
      },
      ':as': {
        'N': stats.archivedSize.toString()
      }
    },
    ReturnValues: 'NONE'
  };
}
